src/frontend/sharedComponents/BottomSheetModal/index.tsx

```diff
@@ -74,34 +80,47 @@ interface Props extends React.PropsWithChildren<{}> {
   onDismiss?: () => void;
   // Triggered by: Android hardware back press and gesture back swipe
   onBack?: () => void;
-  fullHeight?: boolean;
+  fullScreen?: boolean;
 }
 
 export const BottomSheetModal = React.forwardRef<RNBottomSheetModal, Props>(
-  ({children, isOpen, onBack, fullHeight, onDismiss}, ref) => {
+  ({children, isOpen, onBack, onDismiss, fullScreen}, ref) => {
     useBackHandler(isOpen, onBack);
 
+    const {top} = useSafeAreaInsets();
+
+    const contentContextValue = React.useMemo(() => {
+      return {fullScreen: !!fullScreen};
+    }, [fullScreen]);
```

Comment:
nit: I don't think this memoization is necessary. Each modal creates it own instance of a context, so there is no need to worry about changes in state causing unnecessary re-renders of other components

Reply:
addressed via a51326570fa820ddbfda19d3a5cc4d4202c15e46

Reply:
to clarify the initial reasoning: i was worried that having the provider being a child of the BottomSheetModal would potentially be problematic. based on reading the documentation, using the dynamic sizing can cause several re-renders, which i _think_ would affect the children and hence, create a new context value each time.

probably not a big issue in general though